src/client.rs

```diff
@@ -82,6 +93,31 @@ impl VercelBlobClient {
         let token = get_token(self.token_provider.as_deref(), operation, pathname).await?;
         Ok(request.header("authorization", format!("Bearer {}", token)))
     }
+
+    async fn handle_error(response: Response) -> VercelBlobError {
```

Comment:
Can we use a `From<Response.Status>` ?

Reply:
There is already a `From<Status>`:

```
    HttpError(#[from] ReqwestError),
    #[error("Unknown error, please visit https://vercel.com/help ({0}): {1}")]
```

This is handling a `Response`.  I can't do `From<Response>` because this method needs to be async (it has to download the body of the response as JSON to inspect the error code within).

I tried moving the logic here into `error.rs` but I feel it isn't as good of a fit because then some of the API objects (e.g. `BlobApiError`) end up moving out of `client.rs`.

I'm going to leave it as is for now.